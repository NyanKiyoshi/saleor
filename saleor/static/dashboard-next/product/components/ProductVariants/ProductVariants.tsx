import Card, { CardContent } from "material-ui/Card";
import blue from "material-ui/colors/blue";
import green from "material-ui/colors/green";
import red from "material-ui/colors/red";
import { FormControlLabel } from "material-ui/Form";
import Hidden from "material-ui/Hidden";
import { withStyles } from "material-ui/styles";
import Switch from "material-ui/Switch";
import Table, {
  TableBody,
  TableCell,
  TableHead,
  TableRow
} from "material-ui/Table";
import Typography from "material-ui/Typography";
import * as React from "react";

import PageHeader from "../../../components/PageHeader";
import i18n from "../../../i18n";

interface ProductVariantsProps {
  variants: Array<{
    id: string;
    name: string;
    sku: string;
    availability: boolean;
    price: {
      grossLocalized: string;
    };
    grossMargin: string;
  }>;
  onRowClick(id: string);
}

const decorate = withStyles(theme => ({
  alignRightText: {
    textAlign: "right"
  },
  greenText: {
    color: green[500]
  },
  link: {
    color: blue[500],
    cursor: "pointer"
  },
  redText: {
    color: red[500]
  }
}));
export const ProductVariants = decorate<ProductVariantsProps>(
  ({ classes, variants, onRowClick }) => (
    <Card>
      <PageHeader title={i18n.t("Variants")} />
      <Table>
        <TableHead>
          <TableRow>
            <Hidden mdDown>
              <TableCell>{i18n.t("SKU")}</TableCell>
            </Hidden>
            <TableCell>{i18n.t("Name")}</TableCell>
            <TableCell>{i18n.t("Status")}</TableCell>
            <Hidden mdDown>
              <TableCell className={classes.alignRightText}>
                {i18n.t("Price")}
              </TableCell>
              <TableCell className={classes.alignRightText}>
                {i18n.t("Gross margin")}
              </TableCell>
            </Hidden>
          </TableRow>
        </TableHead>
        <TableBody>
          {variants.map(variant => (
            <TableRow key={variant.id}>
              <Hidden mdDown>
                <TableCell>{variant.sku}</TableCell>
              </Hidden>
              <TableCell
                className={classes.link}
                onClick={onRowClick(variant.id)}
              >
                {variant.name}
              </TableCell>
              <TableCell
                className={
                  variant.availability ? classes.greenText : classes.redText
                }
              >
                {variant.availability
                  ? i18n.t("Available")
                  : i18n.t("Unavailable")}
              </TableCell>
              <Hidden mdDown>
                <TableCell className={classes.alignRightText}>
                  {variant.price.grossLocalized}
                </TableCell>
                <TableCell className={classes.alignRightText}>
                  {variant.grossMargin}
                </TableCell>
              </Hidden>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Card>
  )
);
export default ProductVariants;
