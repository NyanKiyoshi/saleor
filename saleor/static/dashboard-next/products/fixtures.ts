export const product = {
  id: "UHJvZHVjdDoyOA==",
  name: "Stewart-Barrera",
  description:
    "Light light office style present life military. Enough executive coach cause own.\n\nCare season wrong fly. Example age between discuss assume.\n\nTheir third build might strong account. Worry enter room note level air.\n\nWeight director mother. Because find task with now enter tree management.\n\nSoon name recently ground green. There owner sport box them red information.",
  collections: {
    edges: [
      {
        node: {
          id: "collection-1",
          name: "Winter Collection"
        }
      },
      {
        node: {
          id: "collection-2",
          name: "Summer Collection"
        }
      }
    ]
  },
  price: {
    localized: "95,47 USD"
  },
  grossMargin: [
    {
      start: 56,
      stop: 98
    }
  ],
  purchaseCost: {
    start: {
      gross: {
        localized: "1,88 USD"
      }
    },
    stop: {
      gross: {
        localized: "48,10 USD"
      }
    }
  },
  priceRange: {
    start: {
      net: {
        localized: "55,14 USD"
      }
    },
    stop: {
      net: {
        localized: "88,36 USD"
      }
    }
  },
  isPublished: true,
  availability: {
    available: true
  },
  images: {
    edges: [
      {
        node: {
          id: "UHJvZHVjdEltYWdlOjY2",
          alt: "",
          order: 0,
          url:
            "/media/products/saleor/static/placeholders/coffee/coffee-02_IS8e8b3.jpg"
        }
      },
      {
        node: {
          id: "UHJvZHVjdEltYWdlOjY3",
          alt: "",
          order: 1,
          url:
            "/media/products/saleor/static/placeholders/coffee/coffee-02_ZIk8H8C.jpg"
        }
      },
      {
        node: {
          id: "UHJvZHVjdEltYWdlOjY4",
          alt: "",
          order: 2,
          url:
            "/media/products/saleor/static/placeholders/coffee/coffee-03_i5b2KpM.jpg"
        }
      }
    ]
  },
  variants: {
    edges: [
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6Nw==",
          sku: "2-1337",
          name: "XXL",
          priceOverride: null,
          stockQuantity: 5
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6OA==",
          sku: "2-1338",
          name: "XL",
          priceOverride: null,
          stockQuantity: 0
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6OQ==",
          sku: "2-1339",
          name: "L",
          priceOverride: null,
          stockQuantity: 15
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6MTA=",
          sku: "2-1340",
          name: "M",
          priceOverride: null,
          stockQuantity: 33
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6MTE=",
          sku: "2-1341",
          name: "S",
          priceOverride: null,
          stockQuantity: 0
        }
      },
      {
        node: {
          id: "UHJvZHVjdFZhcmlhbnQ6MTI=",
          sku: "2-1342",
          name: "XS",
          priceOverride: null,
          stockQuantity: 0
        }
      }
    ]
  },
  productType: {
    id: "UHJvZHVjdFR5cGU6Mw==",
    name: "Coffee"
  },
  url: "/pl/products/stewart-barrera-28/"
};
export const products = [
  {
    id: "UHJvZHVjdDox",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoy",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoz",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  },
  {
    id: "UHJvZHVjdDoa",
    name: "Gardner, Graham and King",
    thumbnailUrl:
      "/media/__sized__/products/saleor/static/placeholders/t-shirts/6_tdo7a5D-crop-c0-5__0-5-255x255-70.jpg",
    productType: {
      id: "1",
      name: "T-Shirt"
    }
  }
];

export const variants = product.variants.edges.map(edge => edge.node);
export const grossMargin = product.grossMargin[0];
export const salePrice = {
  start: product.priceRange.start.net.localized,
  stop: product.priceRange.stop.net.localized
};
export const purchaseCost = {
  start: product.purchaseCost.start.gross.localized,
  stop: product.purchaseCost.stop.gross.localized
};
export const collections = product.collections.edges.map(edge => edge.node);
export const images = product.images.edges.map(edge => edge.node);
