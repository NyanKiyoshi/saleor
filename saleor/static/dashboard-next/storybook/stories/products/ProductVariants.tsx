import { storiesOf } from "@storybook/react";
import * as React from "react";

import ProductVariants from "../../../product/components/ProductVariants";

const variants = [
  {
    id: "1",
    name: "XS",
    sku: "abc123",
    availability: true,
    price: { grossLocalized: "10 $" },
    grossMargin: "5 %"
  },
  {
    id: "2",
    name: "S",
    sku: "abc124",
    availability: true,
    price: { grossLocalized: "10 $" },
    grossMargin: "5 %"
  },
  {
    id: "3",
    name: "M",
    sku: "abc125",
    availability: false,
    price: { grossLocalized: "11 $" },
    grossMargin: "6 %"
  },
  {
    id: "4",
    name: "L",
    sku: "abc126",
    availability: true,
    price: { grossLocalized: "12 $" },
    grossMargin: "6 %"
  }
];

storiesOf("Products / ProductVariants", module)
  .add("when loading data", () => <ProductVariants />)
  .add("when product has no variants", () => <ProductVariants variants={[]} />)
  .add("when product has variants", () => (
    <ProductVariants variants={variants} />
  ))
  .add("with clickable rows", () => (
    <ProductVariants variants={variants} onRowClick={() => {}} />
  ));
