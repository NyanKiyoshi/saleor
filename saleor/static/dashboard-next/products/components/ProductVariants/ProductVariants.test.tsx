import * as React from "react";
import * as renderer from "react-test-renderer";

import { variants } from "../../fixtures";
import ProductVariants from "./";

describe("<ProductVariants />", () => {
  it("renders when loading data", () => {
    const component = renderer.create(<ProductVariants />);
    expect(component).toMatchSnapshot();
  });
  it("renders when product has no variants", () => {
    const component = renderer.create(<ProductVariants variants={[]} />);
    expect(component).toMatchSnapshot();
  });
  it("renders when product has variants", () => {
    const component = renderer.create(<ProductVariants variants={variants} />);
    expect(component).toMatchSnapshot();
  });
  it("renders when product has variants and has clickable rows", () => {
    const component = renderer.create(
      <ProductVariants variants={variants} onRowClick={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
