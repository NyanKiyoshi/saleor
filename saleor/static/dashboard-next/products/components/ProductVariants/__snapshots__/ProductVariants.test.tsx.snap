// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductVariants /> renders when loading data 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
>
  <div
    className="Component-root-35"
  >
    <h2
      className="MuiTypography-root-38 MuiTypography-title-44 Component-title-37"
    >
      Variants
    </h2>
    <div
      className="Component-action-32"
    />
  </div>
  <table
    className="MuiTable-root-62"
  >
    <thead
      className="MuiTableHead-root-63"
    >
      <tr
        className="MuiTableRow-root-64 MuiTableRow-head-65"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          SKU
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Status
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Price
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Gross margin
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-77"
    >
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-skeleton-78"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-skeleton-78"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-skeleton-78"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          <span
            className="Component-skeleton-78"
            style={undefined}
          >
            ‌
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          <span
            className="Component-skeleton-78"
            style={undefined}
          >
            ‌
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<ProductVariants /> renders when product has no variants 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
>
  <div
    className="Component-root-35"
  >
    <h2
      className="MuiTypography-root-38 MuiTypography-title-44 Component-title-37"
    >
      Variants
    </h2>
    <div
      className="Component-action-32"
    />
  </div>
  <table
    className="MuiTable-root-62"
  >
    <thead
      className="MuiTableHead-root-63"
    >
      <tr
        className="MuiTableRow-root-64 MuiTableRow-head-65"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          SKU
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Status
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Price
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Gross margin
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-77"
    >
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          colSpan={2}
          scope={undefined}
        >
          This product has no variants
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<ProductVariants /> renders when product has variants 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
>
  <div
    className="Component-root-35"
  >
    <h2
      className="MuiTypography-root-38 MuiTypography-title-44 Component-title-37"
    >
      Variants
    </h2>
    <div
      className="Component-action-32"
    />
  </div>
  <table
    className="MuiTable-root-62"
  >
    <thead
      className="MuiTableHead-root-63"
    >
      <tr
        className="MuiTableRow-root-64 MuiTableRow-head-65"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          SKU
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Status
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Price
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Gross margin
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-77"
    >
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc123
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          onClick={[Function]}
          scope={undefined}
        >
          XS
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc124
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          onClick={[Function]}
          scope={undefined}
        >
          S
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc125
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          onClick={[Function]}
          scope={undefined}
        >
          M
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc126
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          onClick={[Function]}
          scope={undefined}
        >
          L
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<ProductVariants /> renders when product has variants and has clickable rows 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation2-9 MuiPaper-rounded-6"
>
  <div
    className="Component-root-35"
  >
    <h2
      className="MuiTypography-root-38 MuiTypography-title-44 Component-title-37"
    >
      Variants
    </h2>
    <div
      className="Component-action-32"
    />
  </div>
  <table
    className="MuiTable-root-62"
  >
    <thead
      className="MuiTableHead-root-63"
    >
      <tr
        className="MuiTableRow-root-64 MuiTableRow-head-65"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          SKU
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70"
          scope="col"
        >
          Status
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Price
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-head-70 Component-alignRightText-1"
          scope="col"
        >
          Gross margin
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-77"
    >
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc123
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-link-3"
          onClick={undefined}
          scope={undefined}
        >
          XS
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc124
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-link-3"
          onClick={undefined}
          scope={undefined}
        >
          S
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc125
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-link-3"
          onClick={undefined}
          scope={undefined}
        >
          M
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-64"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          abc126
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-link-3"
          onClick={undefined}
          scope={undefined}
        >
          L
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71"
          scope={undefined}
        >
          <span
            className="Component-redDot-4"
          />
          Unavailable
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-69 MuiTableCell-body-71 Component-alignRightText-1"
          scope={undefined}
        >
          na
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
